### DISCUSSION TITLE
Is it possible to get a .csv of individual tree metrics?

### DISCUSSION THREAD

**Discussion by kalinowskapatrycja** [r-lidar/lasR#10] [Q&A]
Is lasR capable of extracting tree heights and canopy area per tree like in lidR? Or should I be using lidR / analysing just the CHM with a different packages?

**Comment by Jean-Romain** [comment]
`local_maximum` extracts the location of each tree, including the height. `growing_region` is equivalent to `dalponte2016()` and produces a raster of segmented tree. Then, you must analyze the outputs with other tools. There is not built in function for individual tree metrics.

The intentions behind lasR are very different than lidR. lasR tries to provide the most efficient way to process massive point clouds but has a limited set of features. lidR tries to provide the most versatile way to process point clouds, but is weaker at processing large point clouds.

**Reply by kalinowskapatrycja** [reply]
Thank you for your reply

Would it be fair to assume that segment_trees(las, algo) in lidR is more accurate than simply analysying the chm png because it also uses the point clouds in the segmentation? Or am I understanding this wrong 

Also, when copying the tree segmentation code in the lasR tutorial, increasing the local_maximum(20) to something higher than 3 gives me this output. Is this expected?
![image](https://github.com/r-lidar/lasR/assets/95499766/2b86720a-fa2e-4ae0-bbaa-48d6eaf976a3)

**Reply by Jean-Romain** [reply]
> Would it be fair to assume that segment_trees(las, algo) in lidR is more accurate than simply analysying the chm png because it also uses the point clouds in the segmentation? Or am I understanding this wrong.

It really depends on your definition of accurate. It is a strict search of the points that are locally the highest. Is it more accurate than a search on post-processed, pit-free, CHM? I don't know. It depends on you definition of accurate, on what you are searching, on the noise in the point cloud. I'd prefer not to conclude on my side.

> Also, when copying the tree segmentation code in the lasR tutorial, increasing the local_maximum(20) to something higher than 3 gives me this output. Is this expected?

I don't know what I'm seeing. First image seems to be a CHM. Image 2 definitively looks like a bug. Image 3 looks weird. Please report a bug with a reproducible example if you think that something is going wrong.
